docs/tutorials/custom/custom-base-dog.md

```diff
@@ -221,37 +221,51 @@ You can also try turning the robot off and on again, and then retrying the proce
 ## Implement the custom base code
 
 Now that the Freenove server is set up, you will follow the [process for creating modular resources](/operate/get-started/other-hardware/).
-You will use [this module creation tool](https://github.com/viam-labs/generator-viam-module) to simplify the process and generate the necessary stub files.
-Then, you will edit them as necessary to define how each base API method interacts with your robot dog.
+
+### Prerequisites
+
+{{< expand "Install the Viam CLI and authenticate" >}}
+Install the Viam CLI and authenticate to Viam, from the same machine that you intend to upload your module from.
+
+{{< readfile "/static/include/how-to/install-cli.md" >}}
+
+Authenticate your CLI session with Viam using one of the following options:
+
+{{< readfile "/static/include/how-to/auth-cli.md" >}}
+{{< /expand >}}
 
 ### Create the module files
 
-First, from the Raspberry Pi terminal, create a directory inside the home directory to hold your custom code files:
+The CLI module generator generates the files for your modular resource.
 
-```sh {class="command-line" data-prompt="$"}
-mkdir robotdog
-```
+1. Run the following command in your terminal:
 
-Next, install [the module creation tool](https://github.com/viam-labs/generator-viam-module).
-Run `yo viam-module` and follow the prompts to create the files for your custom base.
-As you follow the prompts, it's a good idea to use the same names we used so that your code matches the example code:
+   ```sh {class="command-line" data-prompt="$"}
+   viam module generate
+   ```
 
-- Model name: `robotdog`
-- Module namespace: `viamlabs`
-- Module name: `base`
-- Language: `python`
-- API triplet: `rdk:component:base`
-- Existing API? `Yes`
+2. Follow the prompts, selecting the following options:
 
-{{% alert title="Important" color="note" %}}
+   - Module name: `base`
+   - Language: `Python`
+   - Visibility: `Private`
+   - Namespace/Organization ID:
+     - In the [Viam app](https://app.viam.com), navigate to your organization settings through the menu in upper right corner of the page.
+       Find the **Public namespace** (or create one if you haven't already) and copy that string.
+       In the example snippets below, the namespace is `viamlabs`.
+   - Resource to add to the module (API): `Base Component`.
+   - Model name: `robotdog`
+   - Enable cloud build: `No`
+   - Register module: `Yes`
```

Comment:
Not sure about this since they use it as a local module. Guess it doesn’t matter.